When the new GRKERNSEC_MODSTOP feature is built in and the feature activated by
setting the kernel.grsecurity.disable_modules sysctl, it erroneously locks down
all of the other sysctl/proc tunables too (as would happen when setting
grsec_lock = 1). This was first reported by Alex Efros in gentoo bug #121250.

More info: http://grsecurity.net/pipermail/grsecurity/2006-February/000703.html

--- linux-2.6.14-grsec-2.1.8-2.6.14.6-200601211647/grsecurity/grsec_sysctl.c	2006-02-05 23:18:05.000000000 -0500
+++ linux-2.6.14-grsec-2.1.8-2.6.14.7-200602052251/grsecurity/grsec_sysctl.c	2006-02-05 23:18:35.000000000 -0500
@@ -18,7 +18,8 @@ gr_handle_sysctl_mod(const char *dirname
 	}
 #endif
 #ifdef CONFIG_GRKERNSEC_MODSTOP
-	if (!strcmp(dirname, "grsecurity") && grsec_modstop && (op & 002)) {
+	if (!strcmp(dirname, "grsecurity") && !strcmp(name, "disable_modules") &&
+	    grsec_modstop && (op & 002)) {
 		gr_log_str(GR_DONT_AUDIT, GR_SYSCTL_MSG, name);
 		return -EACCES;
 	}
